### DISCUSSION TITLE
What is the suggested approach for ocr pdf?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2099] [Ideas]
What is the suggested approach for ocr pdf?

**Comment by JorjMcKie** [comment]
Can you please be more specific? Which tool to use? How to decide if at all to OCR? ...
You may want to read this as a start: https://medium.com/p/e455465acb03, also here: https://artifex.com/blog.
Another good source to contact others is PyMuPDF's [Discord](https://discord.gg/TSpYGBW4eq) channel.

**Comment by nissansz** [comment]
Just convert image pdf to text bi-layer pdf with good layout

**Comment by JorjMcKie** [comment]
Try OCRmyPDF. If you want an integration within your PyMuPDF script use the integrated Tesseract access. There are example scripts in the utilities repo.

**Comment by nissansz** [comment]
Tesseract is hard to train and is low in accuracy. I want to change to paddleocr, any method?
Or change to other json based ocr result to format pdf finally.

**Reply by JorjMcKie** [reply]
> Tesseract is hard to train and is low in accuracy. I want to change to paddleocr, any method? Or change to other json based ocr result to format pdf finally.

You obviously know better than me, so please communicate your findings in this channel.

**Comment by nissansz** [comment]
ok
